src/test/java/io/strimzi/kafka/bridge/http/ConsumerSubscriptionIT.java

```diff
@@ -160,6 +161,92 @@ void subscriptionConsumerDoesNotExistBecauseNotCreated(VertxTestContext context)
+        topics.add(topic);
+
+        JsonObject topicsRoot = new JsonObject();
+        topicsRoot.put("topics", topics);
+
+        // Create consumer and subscribe to the topic
+        JsonObject consumerJson = new JsonObject();
+        consumerJson.put("name", name);
+        consumerJson.put("format", "json");
+        consumerService()
+                .createConsumer(context, groupId, consumerJson)
+                .subscribeConsumer(context, groupId, name, topicsRoot);
+
+        // poll to subscribe
+        CompletableFuture<Boolean> consume = new CompletableFuture<>();
+        consumerService()
+                .consumeRecordsRequest(groupId, name, BridgeContentType.KAFKA_JSON_JSON)
+                .as(BodyCodec.jsonArray())
+                .send(ar -> {
+                    if (ar.succeeded()) {
+                        LOGGER.info("Request result: {}", ar.result().body());
+                        consume.complete(true);
+                    }
+                });
+        consume.get(TEST_TIMEOUT, TimeUnit.SECONDS);
+
+        CompletableFuture<Boolean> subscribe = new CompletableFuture<>();
+        // Validate the existing consumer list
+        CompletableFuture<Boolean> listSubscriptions = new CompletableFuture<>();
+        consumerService()
+                .listSubscriptionsConsumerRequest(groupId, name)
+                .as(BodyCodec.jsonObject())
+                .send(ar -> {
+                    context.verify(() -> {
+                        assertThat(ar.succeeded(), is(true));
+                        HttpResponse<JsonObject> response = ar.result();
+                        assertThat(response.statusCode(), is(HttpResponseStatus.OK.code()));
+                        assertThat(response.body().getJsonArray("topics").size(), is(1));
+                    });
+                    listSubscriptions.complete(true);
+                });
+        listSubscriptions.get(TEST_TIMEOUT, TimeUnit.SECONDS);
+
+        // Subscribe to an empty topic list, expect return null response body
+        JsonObject unsubscribeTopicRoot = new JsonObject();
+        unsubscribeTopicRoot.put("topics", new JsonArray());
+        consumerService()
+                .subscribeConsumerRequest(groupId, name, unsubscribeTopicRoot)
+                .sendJsonObject(unsubscribeTopicRoot, ar -> {
+                    context.verify(() -> {
+                        assertThat(ar.succeeded(), is(true));
+                        HttpResponse<JsonObject> response = ar.result();
+                        assertThat(response.statusCode(), is(HttpResponseStatus.NO_CONTENT.code()));
+                        assertThat(response.body(), is(nullValue()));
+                    });
+                    subscribe.complete(true);
+                });
+
+        subscribe.get(TEST_TIMEOUT, TimeUnit.SECONDS);
+        // Validate the existing consumer list
+        CompletableFuture<Boolean> listSubscriptionsAfter = new CompletableFuture<>();
```

Comment:
you can reuse the `listSubscriptions` variable as before just creating a `new CompletableFuture<>();`